### ISSUE TITLE
Is `create_test_data()` public API?

### ISSUE THREAD

**Issue by TomNicholas** [pydata/xarray#2686]
We want to encourage people to use and extend xarray, and we already provide testing functions as public API to help with this.

One function I keep using when writing code which uses xarray is `xarray.tests.test_dataset.create_test_data()`. This is very useful for quickly writing tests for the same reasons that it's useful in xarray's internal tests, but it's not explicitly public API. This means that there's no guarantee it won't change/disappear, which is [not ideal](https://github.com/boutproject/xBOUT/issues/26) if you're trying to write a test suite for separate software. But so many tests in xarray rely on it that presumably it's not going to get changed. 

Is there any reason why it shouldn't be public API? Is there something I should use instead?

**Comment by max-sixty** [comment]
I think we could make that public, interested in others' thoughts

I'm often creating my own ds manually, and should instead just call that to get an object.

We could also promote it to `xarray.testing` if that make it more accessible and the API more explicit. And potentially encourage its usage in tests rather than manually creating a `ds` fixture.

**Comment by jhamman** [comment]
Right, nothing in xarray/tests/ should be considered public API. +1 for moving this to xarray/testing.py if we want to expose this function as public API.
